### PR TITLE
[sw] Clean up boot_rom SPI bootstrap code.

### PR DESCRIPTION
**Pull request by mcy**
This change mostly just adds comments and removes initialisms from
`bootstrap.c`, to make it consistent with the rest of our code.

### CODE REVIEW

File: sw/device/boot_rom/bootstrap.c [hunk 1]
```diff
@@ -33,81 +39,97 @@ static int bootstrap_requested(void) {
   // Read pin
   uint32_t gpio_in;
   dif_gpio_all_read(&gpio, &gpio_in);
-  return !!(gpio_in & GPIO_BOOTSTRAP_BIT_MASK);
+  return gpio_in != 0 && GPIO_BOOTSTRAP_BIT_MASK != 0;
```
Comment (silvestrst) [inline sw/device/boot_rom/bootstrap.c:42]: `return gpio_in != 0 && GPIO_BOOTSTRAP_BIT_MASK != 0;` => `return (gpio_in & GPIO_BOOTSTRAP_BIT_MASK) != 0`.

Would be nice to do it in terms of `bitfield_get_bit32` (which doesn't exist yet).

File: sw/device/boot_rom/bootstrap.c [hunk 2]
```diff
@@ -33,81 +39,97 @@ static int bootstrap_requested(void) {
...
-    result |= hash[i] ^ f->hdr.hash[i];
-  }
-  return result;
+/**
+ * Compares the SHA256 hash of the recieved data with the recieved hash.
+ *
+ * Returns true if the hashes match.
+ */
+static bool check_frame_hash(const spiflash_frame_t *frame) {
+  uint8_t hash[32];
```
Comment (silvestrst) [inline sw/device/boot_rom/bootstrap.c:68]: Could you drop a comment explaining what `32` means?

File: sw/device/boot_rom/bootstrap.c [hunk 3]
```diff
@@ -33,81 +39,97 @@ static int bootstrap_requested(void) {
...
-  for (;;) {
-    if (spid_bytes_available() >= sizeof(f)) {
-      spid_read_nb(&f, sizeof(f));
-      LOG_INFO("Processing frame no: %d exp no: %d", f.hdr.frame_num,
-               expected_frame_no);
-
-      if (FRAME_NO(f.hdr.frame_num) == expected_frame_no) {
-        if (check_frame_hash(&f)) {
-          LOG_ERROR("Detected hash mismatch on frame: %d", f.hdr.frame_num);
+  uint8_t ack[32] = {0};
```
Comment (silvestrst) [inline sw/device/boot_rom/bootstrap.c:82]: `32` again, could it be defined as a named define at the top of the file?

File: sw/device/boot_rom/bootstrap.c (hunk 2 above)
Comment (mcy) [reply sw/device/boot_rom/bootstrap.c:68]: They're both SHA2 hashes. But, sure.

File: sw/device/boot_rom/bootstrap.c (hunk 1 above)
Comment (mcy) [reply sw/device/boot_rom/bootstrap.c:42]: I really really hate how not-type-safe any of this mess is.

File: sw/device/boot_rom/bootstrap.c [hunk 4]
```diff
@@ -4,110 +4,133 @@
...
-               expected_frame_no);
-
-      if (FRAME_NO(f.hdr.frame_num) == expected_frame_no) {
-        if (check_frame_hash(&f)) {
-          LOG_ERROR("Detected hash mismatch on frame: %d", f.hdr.frame_num);
+  uint8_t ack[SHA256_DIGEST_SIZE] = {0};
+  uint32_t expected_frame_num = 0;
+  while (true) {
+    if (spid_bytes_available() >= sizeof(spiflash_frame_t)) {
+      LOG_INFO("Recieving next frame...");
```
Comment (gkelly) [inline sw/device/boot_rom/bootstrap.c:87]: Why add this new message?

File: sw/device/boot_rom/bootstrap.c [hunk 5]
```diff
@@ -4,110 +4,133 @@
...
+      LOG_INFO("Processing frame #%d, expecting #%d", frame_num,
+               expected_frame_num);
+
+      if (frame_num == expected_frame_num) {
+        if (!check_frame_hash(&frame)) {
+          LOG_ERROR("Detected hash mismatch on frame #%d", frame_num);
           spid_send(ack, sizeof(ack));
           continue;
         }
+        LOG_INFO("Hash check successful");
```
Comment (gkelly) [inline sw/device/boot_rom/bootstrap.c:101]: This makes the spiflash process quite noisy with two new messages per frame.

File: sw/device/boot_rom/bootstrap.c [hunk 6]
```diff
@@ -4,110 +4,133 @@
...
 
-        hw_SHA256_hash(&f, sizeof(f), ack);
+        hw_SHA256_hash(&frame, sizeof(spiflash_frame_t), ack);
         spid_send(ack, sizeof(ack));
+        LOG_INFO("SPI ACK successful");
```
Comment (gkelly) [inline sw/device/boot_rom/bootstrap.c:105]: Three new messages per frame?

File: sw/device/boot_rom/bootstrap.c [hunk 7]
```diff
@@ -4,110 +4,133 @@
...
           }
+          LOG_INFO("Flash erase successful");
         }
-        if (flash_write(f.hdr.flash_offset, f.data, ARRAYSIZE(f.data))) {
+
+        if (flash_write(frame.header.flash_offset, frame.data,
+                        SPIFLASH_FRAME_DATA_WORDS) != 0) {
           return E_BS_WRITE;
         }
+        LOG_INFO("Programmed frame #%d", frame_num);
```
Comment (gkelly) [inline sw/device/boot_rom/bootstrap.c:121]: Four new messages per frame?

File: sw/device/boot_rom/bootstrap.c (hunk 4 above)
Comment (mcy) [reply sw/device/boot_rom/bootstrap.c]: Removed them; resolving as discussed.

**Pull request merged**